source/core/TDCodecVia.cpp

```diff
@@ -1074,6 +1074,21 @@ Boolean TDViaParser::EatJSONItem(SubString* input)
     return true;
 }
 
+bool IsInfinityOrNan(SubString token)
+{
+    bool isInfinityOrNan = false;
+    ConstCStr specialIEEEVals[] = { "infinity", "-infinity", "inf", "-inf", "nan", nullptr};
+    ConstCStr* specValPtr = specialIEEEVals;
+    while (*specValPtr) {
+        if (token.ComparePrefixCStrIgnoreCase(*specValPtr)) {
```

Comment:
The behavior of the UnflattenFromJSON node should match LabVIEW's Unflatten From JSON which is stricter about the allowed representations:

![image](https://github.com/user-attachments/assets/69a4f766-ed6a-4efa-b71e-81fb3062e62b)

The TDViaParser is used both for parsing VIA text content and for implementing the UnflattenFromJSON node. We kind of use `Fmt().UseFieldNames()` as a hint that we are doing the UnflattenFromJSON semantics for parsing. We should either use that as a hint to make sure that the UnflattenFromJSON node will align behavior with LabVIEW or add a new hint for this use-case, something like `Fmt().LabVIEWJSONExtensions()` since this is JSON extensions that are [LabVIEW-specific](https://www.ni.com/docs/en-US/bundle/labview-api-ref/page/functions/unflatten-from-json.html).

![image](https://github.com/user-attachments/assets/371a4f56-0642-4002-b347-77da77d3dd1e)